pkg/api/bzz_test.go

```diff
@@ -32,144 +30,395 @@ import (
 	"github.com/ethersphere/bee/pkg/tags"
 )
 
-func TestBzz(t *testing.T) {
+func TestBzzFiles(t *testing.T) {
 	var (
-		bzzDownloadResource = func(addr, path string) string { return "/bzz/" + addr + "/" + path }
-		storer              = smock.NewStorer()
-		ctx                 = context.Background()
-		mockStatestore      = statestore.NewStateStore()
-		logger              = logging.New(ioutil.Discard, 0)
-		client, _, _        = newTestServer(t, testServerOptions{
-			Storer: storer,
+		fileUploadResource   = "/bzz"
+		targets              = "0x222"
+		fileDownloadResource = func(addr string) string { return "/bzz/" + addr }
+		simpleData           = []byte("this is a simple text")
+		mockStatestore       = statestore.NewStateStore()
+		logger               = logging.New(ioutil.Discard, 0)
+		client, _, _         = newTestServer(t, testServerOptions{
+			Storer: smock.NewStorer(),
 			Tags:   tags.NewTags(mockStatestore, logger),
-			Logger: logging.New(ioutil.Discard, 5),
+			Logger: logger,
 		})
-		pipeWriteAll = func(r io.Reader, l int64) (swarm.Address, error) {
-			pipe := builder.NewPipelineBuilder(ctx, storer, storage.ModePutUpload, false)
-			return builder.FeedPipeline(ctx, pipe, r, l)
-		}
 	)
-	t.Run("download-file-by-path", func(t *testing.T) {
-		fileName := "sample.html"
-		filePath := "test/" + fileName
-		missingFilePath := "test/missing"
-		sampleHtml := `<!DOCTYPE html>
-		<html>
-		<body>
-	
-		<h1>My First Heading</h1>
-	
-		<p>My first paragraph.</p>
-	
-		</body>
-		</html>`
 
-		var err error
-		var fileContentReference swarm.Address
-		var fileReference swarm.Address
-		var manifestFileReference swarm.Address
+	t.Run("invalid-content-type", func(t *testing.T) {
+		jsonhttptest.Request(t, client, http.MethodPost, fileUploadResource,
+			http.StatusBadRequest,
+			jsonhttptest.WithRequestBody(bytes.NewReader(simpleData)),
+			jsonhttptest.WithExpectedJSONResponse(jsonhttp.StatusResponse{
+				Message: "invalid content-type header",
```

Comment:
the pattern we usually use is to define the error as a package variable (unexported), then export it using `export_test`, then check for the exact error which is exported in the `export_test_ file in the tests, instead of checking for the hardcoded text value (this allows to change the error messages later on quite easily)

Reply:
I went down a rabbit hole here. There are too many. I tried to do everything in the bzz/dirs files.